src/Interpreters/ThreadStatusExt.cpp

```diff
@@ -537,28 +543,20 @@ void CurrentThread::finalizePerformanceCounters()
     current_thread->finalizePerformanceCounters();
 }
 
-void CurrentThread::detachQuery()
-{
-    if (unlikely(!current_thread))
-        return;
-    current_thread->detachQuery(false);
-}
-
-void CurrentThread::detachQueryIfNotDetached()
+void CurrentThread::detachFromGroupIfNotDetached()
 {
     if (unlikely(!current_thread))
         return;
-    current_thread->detachQuery(true);
+    current_thread->detachFromGroup();
 }
 
-
 CurrentThread::QueryScope::QueryScope(ContextMutablePtr query_context, std::function<void()> fatal_error_callback)
 {
-    CurrentThread::initializeQuery();
-    CurrentThread::attachQueryContext(query_context);
     if (!query_context->hasQueryContext())
         query_context->makeQueryContext();
```

Comment:
That is the reason of performance degradation in that PR.
It happens that the first thread had an incorrect configuration. Query context had been made right after ThreadStatus attaches to the ThreadGroup. As a result no perf events metrics have been gathered from this first thread.

https://github.com/ClickHouse/ClickHouse/pull/52071

Reply:
As follow up I propose turn of metrics_perf_events_enabled for measured runs
https://github.com/ClickHouse/ClickHouse/pull/52072